parser/ini_test.go

```diff
@@ -0,0 +1,436 @@
+		public_ip = true
+		cpu = 4
+		memory = 8192
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		parser.ReadFromString(content)
+		got := parser.WriteToFile("./testdata/writetofile_string_sample.ini")
+		// want := true
+		_, isFile := os.Stat("./testdata/writetofile_string_sample.ini")
+
+		if got != nil {
+			t.Errorf("Got: %v.", got)
+
+		}
+		if isFile != nil {
+			t.Errorf("File wasn't created")
+
+		}
+
+	})
+
+	t.Run("writetofile_fromfile", func(t *testing.T) {
+		parser := NewParser()
+		parser.ReadFromFile(file_path)
+		got := parser.WriteToFile("./testdata/writetofile_sample.ini")
+		// want := true
+		_, isFile := os.Stat("./testdata/writetofile_sample.ini")
+
+		if got != nil {
+			t.Errorf("Got: %v. ", got)
+
+		}
+		if isFile != nil {
+			t.Errorf("File wasn't created")
+
+		}
+	})
+
+}
+
+func TestWrongValues(t *testing.T) {
+
+	t.Run("test_wrong_content_from_string", func(t *testing.T) {
+		parser := NewParser()
+		content := `
+		[Profile
+		name = jarvis
+		password = secret
+		[Deployment]
+		project = peertube
+		name = peertest
+		public_ip = true
+		cpu = 4
+		memory = 8192
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		err := parser.ReadFromString(content)
+
+		if err == nil {
+			t.Errorf("It should've returned an error: %v", err)
+
+		}
+
+	})
+
+	t.Run("test_wrong_content_from_file", func(t *testing.T) {
+		parser := NewParser()
+		err := parser.ReadFromFile(wrong_file_path)
+
+		if err == nil {
+			t.Errorf("It should've returned an error: %v", err)
+
+		}
+
+	})
+}
+
+func TestWrongSection(t *testing.T) {
+
+	t.Run("test_wrong_section_from_string", func(t *testing.T) {
+		parser := NewParser()
+		content := `
+		[Profile]
+		name = jarvis
+		password = secret
+		[Deployment]
+		project = peertube
+		name = peertest
+		public_ip = true
+		cpu = 4
+		memory = 8192
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		err := parser.ReadFromString(content)
+
+		if err != nil {
+			t.Errorf("Error: %v", err)
+
+		}
+
+		_, sectionErr := parser.GetSection("ownerr")
+
+		if sectionErr == nil {
+			t.Errorf("ownerr section exists")
+		}
+	})
+
+	t.Run("test_wrong_section_from_file", func(t *testing.T) {
+		parser := NewParser()
+		err := parser.ReadFromFile(file_path)
+
+		if err != nil {
+			t.Errorf("Error: %v", err)
+
+		}
+
+		_, sectionErr := parser.GetSection("ownerr")
+
+		if sectionErr == nil {
+			t.Errorf("ownerr section exists")
+		}
+	})
+}
+
+func TestWrongGet(t *testing.T) {
+
+	t.Run("test_wrong_key_from_string", func(t *testing.T) {
+		parser := NewParser()
+		content := `
+		[Profile]
+		name = jarvis
+		password = secret
+		[Deployment]
+		project = peertube
+		name = peertest
+		public_ip = true
+		cpu = 4
+		memory = 8192
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		err := parser.ReadFromString(content)
+
+		if err != nil {
+			t.Errorf("Error: %v", err)
+
+		}
+
+		_, getErr := parser.Get("Owner", "namee")
+
+		if getErr == nil {
+			t.Errorf("Key exists")
```

Comment:
the error needs to have a descriptive message and a context e.g the key X in Section Y exists while it shouldn't